src/table/virtualized-table/hooks/use-data/index.ts

```diff
@@ -37,7 +36,7 @@ const useData = (
   const mutableSetCellSize = useMutableProp<SetCellSize>(setCellSize);
   const memoizedToRows = useCallback(
     (qDataPages: EngineAPI.INxDataPage[], prevState?: Row[]) => {
-      const nextState = Array.isArray(prevState) ? [...prevState] : createEmptyState(rowCount);
+      const nextState = Array.isArray(prevState) ? prevState.slice(0, rowCount) : createEmptyState(rowCount);
```

Comment:
could we use `gridState.current.overscanRowStopIndex` (Stop index) instead of `rowCount` or That is not intended for this use case here?

Reply:
No that would not be possible. The "stop index" represent only what react-window has rendered on it's last render. Wheres `rowCount` is the total number of rows in the table (for a page).